test/index.test.js

```diff
@@ -453,11 +448,30 @@ it('generates prod and dev files for files with `process.env.NODE_ENV`', async (
   expect(indexDerivedDev).toContain('false||1')
   expect(indexDerivedProd).toContain('true||1')
 
-  expect(browserDerivedDev).toContain("console.log('esm browser a')")
-  expect(browserDerivedDev).toContain('if (true) {')
-  expect(browserDerivedProd).toContain("console.log('esm browser a')")
-  expect(browserDerivedProd).toContain('if (false) {')
+  let nestedIndexDerivedProd = (
+    await readFile(join(fixtureDir, 'a/index.prod.js'))
+  ).toString()
+  let nesteIndexDerivedDev = (
+    await readFile(join(fixtureDir, 'a/index.dev.js'))
+  ).toString()
+
+  expect(nesteIndexDerivedDev).toContain('if (true) {')
+  expect(nestedIndexDerivedProd).toContain('if (false) {')
+})
+
+it("doesn't process `process.env.NODE_ENV` if index.browser.js is present", async () => {
```

Comment:
What do you think of adding `process.env.NODE_ENV` test to webpack test above?

We got an error with `package.browser` because we tested what package we generate (unit test), but not how our package actually work in webpack (integration test).